solidity/src/processor/libs/ProcessorMessage.sol

```diff
@@ -0,0 +1,367 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.28;
+
+/// @title ProcessorMessage Library
+/// @notice Library for parsing and handling batched messages and subroutines
+/// @dev Handles parsing of complex nested structures from byte arrays
+library ProcessorMessage {
+    /// @notice Type of interval measurement
+    enum IntervalType {
+        Blocks,
+        Seconds
+    }
+
+    /// @notice Contains retry configuration for functions
+    /// @param times Specification of retry attempts
+    /// @param intervalType Type of interval (blocks or seconds)
+    /// @param interval Duration between retry attempts (in blocks or seconds)
+    struct RetryLogic {
+        RetryTimes times;
+        IntervalType intervalType;
+        uint64 interval;
+    }
+
+    /// @notice Type of retry configuration
+    enum RetryTimesType {
+        Indefinitely,
+        Amount
+    }
+
+    /// @notice Specifies how many times to retry
+    /// @param retryType Whether to retry indefinitely or a specific amount
+    /// @param amount Number of times to retry if type is Amount
+    struct RetryTimes {
+        RetryTimesType retryType;
+        uint64 amount;
```

Comment:
if RetryTimesType is `Indefinitely`, will amount be 0 and just be ignored?

Reply:
This will all be reworked but yes, the idea was to send something there and ignore it.